test/test_euler.py

```diff
@@ -91,22 +92,32 @@ def rand():
     energy = rand()
     mom = make_obj_array([rand() for _ in range(dim)])
 
-    q = join_conserved(dim, mass=mass, energy=energy, momentum=mom)
+    mass_fractions = make_obj_array([rand() for _ in range(nspecies)])
+    species_mass = mass * mass_fractions
+
+    q = join_conserved(dim, mass=mass, energy=energy, momentum=mom,
+                       species_mass=species_mass)
     cv = split_conserved(dim, q)
 
     # {{{ create the expected result
 
     p = eos.pressure(cv)
     escale = (energy + p) / mass
 
-    expected_flux = np.zeros((dim + 2, dim), dtype=object)
+    numeq = dim + 2 + nspecies
+
+    expected_flux = np.zeros((numeq, dim), dtype=object)
     expected_flux[0] = mom
     expected_flux[1] = mom * escale
 
     for i in range(dim):
         for j in range(dim):
             expected_flux[2+i, j] = (mom[i] * mom[j] / mass + (p if i == j else 0))
 
+            #    if nspecies > 1:
+    for i in range(nspecies):
+        expected_flux[dim+2+i] = mom * make_obj_array([mass_fractions[i] / mass])
```

Comment:
```suggestion
        expected_flux[dim+2+i] = mom * make_obj_array([species_mass[i] / mass])
```
Right? (I don't know what it's actually supposed to be, just comparing with euler.py.)

The `make_obj_array([...])` can be omitted too.

Reply:
Yes, good catch.